### PR TITLE
Respond to the Configuration update

### PR DESCRIPTION
**Pull request by troydai**
Decouples from DependencyInjection. Rather than searching for
IApplicationEnvironment instance, the BasePath on IConfigurationSourceRoot
is used to locate user secrets when user secrets id is not given.

https://github.com/aspnet/Configuration/pull/175

### CODE REVIEW

File: test/SecretManager.Tests/ConfigurationExtensionTests.cs [hunk 1]
```diff
@@ -28,13 +28,7 @@ public void AddUserSecrets_Does_Not_Fail_On_Non_Existing_File()
             var backupServiceProvider = CallContextServiceLocator.Locator.ServiceProvider;
```
Comment (Praburaj) [inline test/SecretManager.Tests/ConfigurationExtensionTests.cs]: You can remove backing up the `ServiceProvider` and also remove the finally block.

Comment (Praburaj) [inline test/SecretManager.Tests/ConfigurationExtensionTests.cs]: Same comment on both the tests.

Comment (Praburaj) [inline test/SecretManager.Tests/ConfigurationExtensionTests.cs]: Also remove the `MyApplicationEnvironment` class if not needed.

**Comment by Praburaj** [comment]
:shipit: after addressing these comments.

**Comment by Praburaj** [comment]
Can you also remove the `MyApplicationEnvironment` class from this test file?

**Comment by troydai** [comment]
OK

**Comment by Praburaj** [comment]
:shipit: after that.

**Comment by Praburaj** [comment]
@troydai can you also remove the DI dependency in the test project.json? 

https://github.com/aspnet/UserSecrets/blob/dev/test/SecretManager.Tests/project.json#L4

**Comment by troydai** [comment]
Yep, nice catch.

**Pull request merged**